### PR TITLE
Saghmrossi/refactor

### PR DESCRIPTION
**Pull request by saghmrossi**
A lot of my recent pull requests had missing and/or outdated documentation and some moderate style issues, so I made a separate branch to patch things up a bit. It was checked out from saghmrossi/update, and is currently up to date with it, but should probably not be merged into 1.0 until the other branch merged.

### CODE REVIEW

File: src/client/cursor.rs [hunk 1]
```diff
@@ -103,12 +147,23 @@ impl <'a, T> Cursor<'a, T> where T: Read + Write + 'a {
...
+    /// # Return value
+    ///
+    /// Returns the newly-created method.
     fn new_get_more_request(&mut self) -> Message {
         Message::with_get_more(self.request_id,
                                self.namespace.to_owned(),
                                self.batch_size, self.cursor_id)
     }
 
+    /// Attempts read another batch of BSON documents from the stream.
```
Comment (alabid) [inline src/client/cursor.rs]: s/Attempts/Attempts to

File: src/client/wire_protocol/operations.rs [hunk 2]
```diff
@@ -109,14 +136,25 @@ impl Message {
 
         let header = Header::with_update(total_length, request_id);
 
-        Ok(Message::OpUpdate { header: header,
-                               namespace: namespace,
+        Ok(Message::OpUpdate { header: header, namespace: namespace,
                                flags: flags, selector: selector,
                                update: update })
-   }
+    }
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: Good.

File: src/client/wire_protocol/operations.rs [hunk 3]
```diff
@@ -181,7 +219,9 @@ impl Message {
         let option_length = match return_field_selector {
             Some(ref bson) => match bson.byte_length() {
                 Ok(i) => i,
-                Err(_) => return Err("Unable to serialize return_field_selector".to_owned())
+                Err(_) =>
+                    return Err("Unable to serialize \
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: I'd prefer not to include a line break within a string unless absolutely necessary. In this case, we can fit the string in one line.

File: src/client/wire_protocol/operations.rs [hunk 4]
```diff
@@ -358,12 +481,11 @@ impl Message {
         };
 
         match return_field_selector {
-            &Some(ref bson) => match Message::write_bson_document(buffer, bson) {
+            &Some(ref b) => match Message::write_bson_document(buffer, b) {
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: Avoid one letter variable names.

File: src/client/wire_protocol/operations.rs [hunk 5]
```diff
@@ -294,6 +362,60 @@ impl Message {
...
+        }
+
+        // Writes the null terminator for the collection name string.
+        match buffer.write_u8(0) {
+            Ok(_) => (),
+            Err(_) => return Err("Unable to write namespace".to_owned())
+        };
+
+
+        for bson in documents {
```
Comment (alabid) [inline src/client/wire_protocol/operations.rs]: "for doc in documents"

File: src/client/wire_protocol/operations.rs (hunk 4 above)
Comment (saghmrossi) [inline src/client/wire_protocol/operations.rs]: Is "bson" considered a bad variable name here as well as the previous case?

Comment (alabid) [inline src/client/wire_protocol/operations.rs]: Yes. I believe you changed the use of "bson" in the previous case to "doc"

**Comment by alabid** [comment]
I just have a couple of comments. The refactor looks good so far.

File: src/client/wire_protocol/header.rs [hunk 6]
```diff
@@ -102,20 +107,54 @@ impl Header {
         Header::new(message_length, request_id, 0, op_code)
     }
 
+    /// Construcs a new Header for a reply.
+    ///
+    /// # Arguments
+    ///
+    /// `message_length` - The length of the message in bytes.
+    /// `request_id` - Identifier for the request, or `0` if the the message
```
Comment (vkarpov15) [inline src/client/wire_protocol/header.rs]: Let's be consistent on spacing - if we're lining up `-` horizontally, let's do it for all of the options

Comment (saghmrossi) [inline src/client/wire_protocol/header.rs]: I was just following the example in the style guideline of lining up each line of a single argument but not the arguments themselves: https://github.com/rust-lang/rust/blob/master/src/doc/style/style/comments.md

Comment (vkarpov15) [inline src/client/wire_protocol/header.rs]: Looks like `request_id` is lined up with `op_code` below though, let's tweak that for consistency.

Comment (saghmrossi) [inline src/client/wire_protocol/header.rs]: Oops, looks like I did that a couple times. Not sure how that happened.

**Pull request merged**